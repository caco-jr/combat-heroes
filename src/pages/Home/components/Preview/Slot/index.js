import React from 'react';

import { PreviewSlotWrapper } from './index.style';

const PreviewSlot = ({ name }) => {
  return <PreviewSlotWrapper>{name}</PreviewSlotWrapper>;
};

export default PreviewSlot;
