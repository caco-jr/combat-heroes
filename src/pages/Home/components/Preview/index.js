import React from 'react';
import { PreviewWrapper, PreviewBackground } from './index.style';
import PreviewSlot from './components/Slot';

const Preview = () => {
  const slots = [
    {
      response: 'success',
      id: '70',
      name: 'Batman',
      powerstats: {
        intelligence: '100',
        strength: '26',
        speed: '27',
        durability: '50',
        power: '47',
        combat: '100'
      },
      biography: {
        'full-name': 'Bruce Wayne',
        'alter-egos': 'No alter egos found.',
        aliases: ['Insider', 'Matches Malone'],
        'place-of-birth': 'Crest Hill, Bristol Township; Gotham County',
        'first-appearance': 'Detective Comics #27',
        publisher: 'DC Comics',
        alignment: 'good'
      },
      appearance: {
        gender: 'Male',
        race: 'Human',
        height: ["6'2", '188 cm'],
        weight: ['210 lb', '95 kg'],
        'eye-color': 'blue',
        'hair-color': 'black'
      },
      work: {
        occupation: 'Businessman',
        base:
          'Batcave, Stately Wayne Manor, Gotham City; Hall of Justice, Justice League Watchtower'
      },
      connections: {
        'group-affiliation':
          'Batman Family, Batman Incorporated, Justice League, Outsiders, Wayne Enterprises, Club of Heroes, formerly White Lantern Corps, Sinestro Corps',
        relatives:
          'Damian Wayne (son), Dick Grayson (adopted son), Tim Drake (adopted son), Jason Todd (adopted son), Cassandra Cain (adopted ward)\nMartha Wayne (mother, deceased), Thomas Wayne (father, deceased), Alfred Pennyworth (former guardian), Roderick Kane (grandfather, deceased), Elizabeth Kane (grandmother, deceased), Nathan Kane (uncle, deceased), Simon Hurt (ancestor), Wayne Family'
      },
      image: {
        url: 'https://www.superherodb.com/pictures2/portraits/10/100/639.jpg'
      }
    },
    {
      response: 'success',
      id: '47',
      name: 'Astro Boy',
      powerstats: {
        intelligence: 'null',
        strength: '90',
        speed: 'null',
        durability: 'null',
        power: 'null',
        combat: 'null'
      },
      biography: {
        'full-name': 'Atom',
        'alter-egos': 'No alter egos found.',
        aliases: ['-'],
        'place-of-birth': '-',
        'first-appearance': '-',
        publisher: 'null',
        alignment: 'good'
      },
      appearance: {
        gender: 'Male',
        race: 'null',
        height: ['-', '0 cm'],
        weight: ['- lb', '0 kg'],
        'eye-color': 'Brown',
        'hair-color': 'Black'
      },
      work: { occupation: '-', base: '-' },
      connections: { 'group-affiliation': '-', relatives: '-' },
      image: {
        url: 'https://www.superherodb.com/pictures2/portraits/10/100/1159.jpg'
      }
    }
  ];

  return (
    <PreviewBackground>
      <PreviewWrapper className="container">
        {slots.map(item => (
          <PreviewSlot
            key={item.id}
            name={item.name}
            biography={item.biography}
            image={item.image}
          />
        ))}
      </PreviewWrapper>
    </PreviewBackground>
  );
};

export default Preview;
