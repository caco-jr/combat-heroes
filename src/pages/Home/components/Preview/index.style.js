import styled from 'styled-components';

export const PreviewBackground = styled.section`
  background: linear-gradient(
    90deg,
    var(--secondary-color) 50%,
    var(--primary-color) 50%
  );
`;

export const PreviewWrapper = styled.section`
  position: relative;
  display: flex;

  section {
    flex: 1;
    min-height: 400px;
    padding: 30px;
  }
`;
