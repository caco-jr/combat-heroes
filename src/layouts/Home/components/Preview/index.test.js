import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Provider } from 'react-redux';

import Preview from './index';
import makeStore from '../../../../redux';

describe('With Enzyme', () => {
  it('renders', () => {
    const wrapper = shallow(
      <Provider store={makeStore()}>
        <Preview />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe('With Snapshot Testing', () => {
  it('Rendering Preview', () => {
    const component = renderer.create(
      <Provider store={makeStore()}>
        <Preview />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
