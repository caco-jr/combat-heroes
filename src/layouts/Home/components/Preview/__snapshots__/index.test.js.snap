// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Enzyme renders 1`] = `ShallowWrapper {}`;

exports[`With Snapshot Testing Rendering Preview 1`] = `
.c0 {
  background: linear-gradient( 90deg,var(--secondary-color) 50%,var(--primary-color) 50% );
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 section {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-height: 400px;
  padding: 30px;
}

<section
  className="c0"
>
  <section
    className="c1 container"
  >
    <section
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Aleatório
      </button>
    </section>
    <section
      className=""
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Aleatório
      </button>
    </section>
  </section>
</section>
`;
